### PR TITLE
Adds GitHub flow to automate busybox container SHAs

### PR DESCRIPTION
**Pull request by wiardvanrij**
Signed-off-by: Rij, W. van (Wiard) <wiard@outlook.com>

Relates to #4949 

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

The .github workflow does a cronjob task. It checks Quay for the latest Prometheus/busybox and fetches the SHA's for all platforms. These values are placed as bash variables in `.busybox-versions`. That file is included in the Makefile so the vars can be used as build arguments.

The cronjob will make a PR against `main`, if there is a change. 

Consider changing the cronjob to do it every 2-3 days, and allow manual updates if needed (?).

The reason we pin the version is that we want to know which version we are using AND that builds are consistent. Not just for the main branch but as well for local development, etc.

## Verification

I have tested this in my own fork. Which did work. I assume (ye ye) it should work here as well out of the box. I can't test if we have the proper rights, but it _should_ be fine as I did not use any special things here.

### CODE REVIEW

File: .github/workflows/container-version.yaml
Comment (saswatamcode) [inline .github/workflows/container-version.yaml:3]: Why at 13:37 daily?

File: .github/workflows/container-version.yaml
Comment (saswatamcode) [inline .github/workflows/container-version.yaml:18]: Maybe we can add `branch: update-sha-action` and `delete-branch: true` here so that the branches are auto-deleted after merge and we know what branches are made by this action?

File: .github/workflows/container-version.yaml
Comment (saswatamcode) [inline .github/workflows/container-version.yaml:15]: How about adding a step after this to log the PR link using this: https://github.com/peter-evans/create-pull-request#action-outputs? It can then be referencable from [Action run](https://github.com/thanos-io/thanos/actions). 🙂

**Review by saswatamcode — APPROVED** [review]
This looks awesome! 💫
Just some questions/suggestions.

File: .github/workflows/container-version.yaml
Comment (wiardvanrij) [reply .github/workflows/container-version.yaml:3]: It's just a daily check. 1337 is just a random time :p 
We should see how it works/runs and adjust cron to our likings.

File: .github/workflows/container-version.yaml
Comment (wiardvanrij) [reply .github/workflows/container-version.yaml:18]: It creates a patch, so I believe there is less of a mess in PR style. We can further check things by the PR's it's making as reference if needed. If this behaviour needs to be changed we can iterate on it. 
Basically it's now in the style of dependabot.

File: .github/workflows/container-version.yaml
Comment (wiardvanrij) [reply .github/workflows/container-version.yaml:15]: Yea I'm not sure what you mean. 
What I do think is that we should move this forward, and you make an iteration with your magic on it. You have some good ideas and know how to make this better. No need to be limited by my work :) WDYT?

File: .github/workflows/container-version.yaml
Comment (matej-g) [reply .github/workflows/container-version.yaml:3]: Yeah, random :wink:

File: .github/workflows/container-version.yaml
Comment (matej-g) [inline .github/workflows/container-version.yaml:25]: Looks like we can add `team-reviewers` as well, maybe it would be worth to add Thanos maintainers or so, so we make sure these PRs do not slip through?

File: scripts/busybox-updater.sh
Comment (matej-g) [inline scripts/busybox-updater.sh:7]: Nit: maybe specify what each call is doing e.g.:
```suggestion
echo "Getting latest tags"
```

File: .github/workflows/container-version.yaml
Comment (matej-g) [inline .github/workflows/container-version.yaml:21]: Nit: PR titles usually follow convention `<component>: <title>`, maybe
```suggestion
          title: '*: Updates Prometheus BusyBox image SHAs'
```

**Review by matej-g — APPROVED** [review]
Nicely done :raised_hands: Few optional suggestion, but it's already good to go :+1:

File: .github/workflows/container-version.yaml
Comment (saswatamcode) [reply .github/workflows/container-version.yaml:15]: Sure, we can totally iterate! Would be nice to also get community input as well once this is in action! 🤗

File: .github/workflows/container-version.yaml
Comment (wiardvanrij) [reply .github/workflows/container-version.yaml:25]: We will need a private access token to enable this.

File: .github/workflows/container-version.yaml
Comment (wiardvanrij) [reply .github/workflows/container-version.yaml:18]: implemented

File: Makefile
Comment (yeya24) [inline Makefile:440]: Please add a new line

File: .github/workflows/container-version.yaml
Comment (yeya24) [reply .github/workflows/container-version.yaml:27]: Please add a new line here.

**Review by yeya24 — DISMISSED** [review]
LGTM and some nits only. Thanks!

**Comment by wiardvanrij** [comment]
I currently don't have a workstation anymore, figured to manually edit those things via GitHub online. Seems like it won't sign my commits then.. :( 
This might take a bit until I have set up a workplace with proper ssh/gpg etc again. Sorry about that.

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**
